scrapy/settings/__init__.py

```diff
@@ -110,6 +139,25 @@ def setmodule(self, module, priority='project'):
             if key.isupper():
                 self.set(key, getattr(module, key), priority)
 
+    def update(self, values, priority='project'):
+        self._assert_mutability()
+        if values is not None:
+            if isinstance(values, BaseSettings):
+                for name, value in six.iteritems(values):
+                    self.set(name, value, values.getpriority(name))
+            else:
+                for name, value in six.iteritems(values):
+                    self.set(name, value, priority)
+
+    def delete(self, name, priority='project'):
+        self._assert_mutability()
+        priority = get_settings_priority(priority)
+        if priority >= self.getpriority(name):
+            del self.attributes[name]
+
+    def __delitem__(self, name):
+        self.delete(name)
```

Comment:
I like having the option of "Delete this setting *unless someone has updated it with a higher priority than what I'm doing*", but would yield to the majority if that is considered useless. ;)
But I think you're right in that `__delitem__` should ignore priorities and just delete from `attributes`